---
title: "Nitrifying consortium GitHub codebook"
output: html_notebook
editor_options: 
  chunk_output_type: inline
---
# Table of contents
- Figure 1: Visualize the relative abundance and metabolic data for the combined PacBio read set on the starting consortium material
- Figure S1: Visualize the relative abundance and metabolic data for the PacBio read set generated with different DNA extraction kits used on the starting consortium material
- Figure S2: Optical density and pH data for enrichment cultures grown along a redox gradient representative of WWTPs
- Figure 2: Nitrogen chemistry analyses
- Figure 3: Beta diversity analyses using non-metric multidimensional scaling for metagenomic DNA sequences obtained from enrichment culture samples
- Figure S3: Beta diversity analyses using non-metric multidimensional scaling for metagenomic DNA sequences obtained from enrichment culture samples omitting aerobic cultures
- Figure 4: Relative abundance for MAG >= 1% from enrichment cultures grown along a redox gradient
- Figure S4: RNA read count data for key nitrogen cycling genes from enrichment cultures
- Figure S5: Relative abundance for all MAGs (no abundance cutoff) from enrichment cultures grown along a redox gradient
- Figure S6: Percent of RNA reads mapped to genome bins obtained from the incubation time series
- Figure S7: Top 25 values for RNA reads mapped to amino acid metabolic pathways

# Figure 1: Visualize the relative abundance and metabolic data for the combined PacBio read set on the starting consortium material

1. Load the required files including the Sylph output and trim it down to the column that contains the combined data for relative abundance.
```{r}
#Load the required libraries for data manipulation
library(tidyverse)
library(readxl)
library(data.table)
library(glue)
library(ggtext)
library(ggh4x)
library(patchwork)

#Load the relative abundance and metabolic data from the joint data frame
Long_read_joint_file_sylph_DRAM <- fread("Fig1_relative_abundance_and_nitrogen_metabolism_PacBio_combined_reads_data.tsv", sep = "\t", header = "auto")

#Write the table to provie source data for figure
write.table(Long_read_joint_file_sylph_DRAM, file = "Fig1_relative_abundance_and_nitrogen_metabolism_PacBio_combined_reads_data.tsv", sep = "\t", row.names = FALSE)
```

2. Run the gather function as needed to convert data frames from wide to long formats for visualization
```{r}
Long_read_MAGs_sylph_percent_complete_gathered <- Long_read_joint_file_sylph_DRAM %>% 
                                                  gather(key = Pathway_name, value = Percent_complete, 
                                                        "3-Hydroxypropionate bi-cycle":"Complex V: V/A-type ATPase, prokaryotes", na.rm=FALSE) %>%
                                                  select(-c("CAZy: Alpha-galactans": "Sulfur metabolism: thiosulfate => sulfite")) %>% 
                                                  mutate(group = case_when(grepl("\\:.*", Pathway_name) ~ 
                                                  sub("\\:.*", "", Pathway_name),
                                                  !grepl("\\:.*", Pathway_name) ~ "Module")) %>% #maintained up to this step
                                                  mutate(Pathway_name = gsub("Complex I*:","",Pathway_name)) %>% #had to be more specific in filtering
                                                  mutate(Pathway_name = gsub("Complex IV*.*:","",Pathway_name)) %>%
                                                  mutate(Pathway_name = gsub("Complex V*.*:","",Pathway_name))

Long_read_MAGs_sylph_true_false_gathered<- Long_read_joint_file_sylph_DRAM %>%
                                           gather(Pathway_name, Presence_absence,
                                                  "CAZy: Alpha-galactans":"Sulfur metabolism: thiosulfate => sulfite", na.rm = FALSE) %>%
                                           select(-c("3-Hydroxypropionate bi-cycle":"Complex V: V/A-type ATPase, prokaryotes")) %>%
                                           mutate(group = case_when(grepl("\\:.*", Pathway_name) ~ 
                                                  sub("\\:.*", "", Pathway_name))) %>%
                                           mutate(Pathway_name = gsub(".*:","",Pathway_name))

#Clean up the data and reorder
Long_read_MAGs_sylph_percent_complete_gathered$Pathway_name<-trimws(Long_read_MAGs_sylph_percent_complete_gathered$Pathway_name, which = c("left")) 

Long_read_MAGs_sylph_percent_complete_gathered$Pathway_name = factor(Long_read_MAGs_sylph_percent_complete_gathered$Pathway_name, 
                                          levels =c(
                                            
                                          #Module facet
                                            
                                          "Glycolysis (Embden-Meyerhof pathway), glucose => pyruvate", 
                                          "Pentose phosphate pathway (Pentose phosphate cycle)",
                                          "Entner-Doudoroff pathway, glucose-6P => glyceraldehyde-3P + pyruvate",
                                          "Citrate cycle (TCA cycle, Krebs cycle)",
                                          "Glyoxylate cycle",
                                          "Reductive pentose phosphate cycle (Calvin cycle)",
                                          "Reductive citrate cycle (Arnon-Buchanan cycle)",
                                          "Dicarboxylate-hydroxybutyrate cycle",
                                          "Hydroxypropionate-hydroxybutylate cycle",
                                          "3-Hydroxypropionate bi-cycle",
                                          "Reductive acetyl-CoA pathway (Wood-Ljungdahl pathway)",
                                          "Acetyl-CoA pathway, CO2 => acetyl-CoA",
                                          "Methanogenesis, CO2 => methane",
                                          
                                          #Complex I
                                          
                                          "NADH:quinone oxidoreductase, prokaryotes",
                                          "NAD(P)H:quinone oxidoreductase, chloroplasts and cyanobacteria",
                                          "NADH dehydrogenase (ubiquinone) 1 alpha subcomplex",
                                          
                                          #Complex II
                                          
                                          "Succinate dehydrogenase (ubiquinone)",
                                          "Succinate dehydrogenase, prokaryotes",
                                          "Fumarate reductase, prokaryotes",
                                          
                                          #Complex III
                                          
                                          "Cytochrome bc1 complex respiratory unit", #what's the difference between first and second?
                                          "Cytochrome bc1 complex",
                                          "Cytochrome bd ubiquinol oxidase",
                                          
                                          #Complex IV high affinity
                                          
                                          #"Cytochrome bd ubiquinol oxidase", This level is being duplicated, but may not be an issue
                                          "Cytochrome c oxidase, cbb3-type",
                                          
                                          #Complex IV low affinity
                                          
                                          "Cytochrome c oxidase",
                                          "Cytochrome c oxidase, prokaryotes",
                                          "Cytochrome aa3-600 menaquinol oxidase",
                                          "Cytochrome o ubiquinol oxidase",
                                          
                                          #Complex V
                                          "F-type ATPase, prokaryotes and chloroplasts",
                                          "F-type ATPase, eukaryotes",
                                          "V/A-type ATPase, prokaryotes",
                                          "V-type ATPase, eukaryotes"
                                          ))

#Categorical data script
Long_read_MAGs_sylph_true_false_gathered$Pathway_name<-trimws(Long_read_MAGs_sylph_true_false_gathered$Pathway_name, which = c("left")) 

Long_read_MAGs_sylph_true_false_gathered$Pathway_name = factor(Long_read_MAGs_sylph_true_false_gathered$Pathway_name, 
                                          levels =c(
                                            
                                          #CAZy facet
                                            
                                          "Polyphenolics", 
                                          "Crystalline Cellulose",
                                          "Amorphous Cellulose",
                                          "Xyloglucan",
                                          "Xylans",
                                          "Alpha-mannan",
                                          "Beta-mannan",
                                          "Mixed-Linkage glucans",
                                          "Starch",
                                          "Pectin",
                                          "Arabinan",
                                          "Alpha-galactans",
                                          "Beta-galactan (pectic galactan)",
                                          "Chitin",
                                          "Sulf-Polysachharides",
                                          "Mucin",
                                          "Fucose Cleavage",
                                          "Arabinose cleavage",
                                          "Rhamnose cleavage",
                                          
                                          #Nitrogen metabolism
                                          
                                          "ammonia => nitrite",
                                          "Bacterial/Archaeal ammonia oxidation",
                                          "Bacterial (aerobic-specific) ammonia oxidation",
                                          "Bacterial (anaerobic-specific) ammonia oxidation",
                                          "nitrite => nitrate",
                                          "nitrate => nitrite",
                                          "nitrite => nitric oxide",
                                          "nitric oxide => nitrous oxide",
                                          "nitrous oxide => nitrogen",
                                          "nitrogen => ammonia",
                                          "Dissimilatory nitrite reduction to ammonia (DNRA)",
                                          "Nitrogen fixation alternative", #typo corrected higher up
                                          
                                          #Sulfur metabolism
                                          "dissimilatory sulfate reduction (and oxidation) sulfate => sulfide",
                                          "thiosulfate => sulfite",
                                          "Thiosulfate oxidation by SOX complex, thiosulfate => sulfate",
                                          "tetrathionate => thiosulfate",
                                          
                                          #Other reductases
                                          
                                          "selenate/Chlorate reduction",
                                          "TMAO reductase",
                                          "mercury reduction",
                                          "arsenate reduction, pt 1",
                                          "arsenate reduction, pt 2",
                                          
                                          #Photosynthesis
                                          "Photosystem I",
                                          "Photosystem II",
                                          
                                          #Methanogenesis and methanotrophy
                                          
                                          "Key functional gene",
                                          "acetate => methane, pt 1",
                                          "acetate => methane, pt 2",
                                          "acetate => methane, pt 3",
                                          "methanol => methane",
                                          "trimethylamine => dimethylamine",
                                          "dimethylamine => monomethylamine",
                                          "monomethylamine => ammonia",
                                          "putative but not defining CO2 => methane",
                                          "methane => methanol, with oxygen (pmo)",
                                          "methane => methanol, with oxygen (mmo)",
                                          
                                          #SCFA and alcohol conversions
                                          "pyruvate => acetyl CoA v1",
                                          "pyruvate => acetyl CoA v2",
                                          "pyruvate => acetylCoA f+ formate v3",
                                          "acetate, pt 1",
                                          "acetate, pt 2",
                                          "acetate, pt 3",
                                          "lactate L",
                                          "lactate D",
                                          "Butyrate, pt 1",
                                          "Butyrate, pt 2",
                                          "Propionate, pt 1",
                                          "Propionate, pt 2",
                                          "Alcohol production"
                                          ))
```

3. Generate a relative abundance heatmap with a single column that will be joined to the DRAM data afterwards.
```{r}
heatmap_theme<-  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
                 panel.background = element_blank(), axis.line = element_line(colour = "black"),
                 axis.text.x = element_text(color = "black", size = 14),
                 axis.text.y = element_text(color = "black", size = 14),
                 axis.title.x = element_text(color = "black", size = 16, vjust = -3),
                 axis.title.y = element_text(color = "black", size = 38, vjust = 6),
                 legend.title=element_text(size=32, vjust = 2), 
                 legend.text=element_text(size=30),
                 legend.key = element_rect(fill = "white", color = NA),
                 plot.title = element_text(size = 40, face = "bold", hjust = 0.5),
                 panel.border = element_rect(colour = "black", fill=NA, size=1),
                 plot.margin = unit(c(2, 2, 2, 2), "cm"))

Starting_material_relative_abundance_heatmap <- Long_read_MAGs_sylph_percent_complete_gathered %>%
                                                mutate_at(c('Relative_abundance'), ~na_if(., 0)) %>% #convert 0 to NA to facilitate visualization
                                                filter(!is.na(Relative_abundance)) %>% #need to re add this filter for it to work
                                                ggplot(aes(x= Kingdom, y = reorder(Family_genus_bin, Relative_abundance), fill= Relative_abundance)) +
                                                geom_tile(aes()) + #use a dummy variable Kingdom to limit x entry to a single column
                                                theme(axis.text.x=element_text(angle=0, hjust=1),
                                                axis.title.y = element_text(color = "black", size = 24, vjust = 0),
                                                legend.key.size = unit(2, 'cm'))+
                                                scale_fill_viridis_c(name="Relative abundance (%)", 
                                                                     na.value = "white", limits = c(0, 50)) + #set to max which is just above 80
                                                labs(title = "",
                                                x = "" , y = "MAG ID") +
                                                heatmap_theme +
                                                theme(axis.text.x = element_blank(),
                                                      axis.text.y = element_markdown(size = 38, face = "bold"),
                                                      axis.title.x = element_blank(),
                                                      axis.title.y = element_text(color = "black", size = 44, vjust = 6, face = "bold"),
                                                      strip.text.x = element_text(color = "black", size = 40, face = "bold"),
                                                      legend.title=element_text(size=48, vjust = 2, face = "bold"), 
                                                      legend.text=element_text(size=44, face = "bold"),
                                                      plot.margin=unit(c(0, -2, 0, 0), "cm"))
```

4. Generate the nitrogen metabolism heat map and connect everything together.
```{r}
#recode the TRUE/FALSe data to presence absence
Long_read_MAGs_sylph_true_false_gathered<-Long_read_MAGs_sylph_true_false_gathered %>% 
                                           mutate(Presence_absence = as.factor(Presence_absence)) %>%
                                           mutate(Presence_absence = recode(Presence_absence, 'TRUE' = "Present", 'FALSE' = "Absent"))

Long_read_MAGs_sylph_nitrogen_metabolism_heatmap<- Long_read_MAGs_sylph_true_false_gathered %>%
                                                    mutate_at(c('Relative_abundance'), ~na_if(., 0)) %>% #convert 0 to NA to facilitate visualization
                                                    filter(!is.na(Relative_abundance)) %>%
                                                    filter(group == "Nitrogen metabolism") %>%
                                                    ggplot(aes(x=Pathway_name, y = reorder(Family_genus_bin, Relative_abundance),
                                                               fill=Presence_absence))+
                                                    geom_tile(aes())+
                                                    facet_grid(~ group, 
                                                             scales = "free", space = "free") +
                                                    theme(axis.text.x=element_text(angle=90, hjust=1)) +
                                                    scale_fill_manual(values = c("#CCFFFF", "#009933"))+ 
                                                    labs(title = "", x = "", y = "", fill = "Function") +
                                                    heatmap_theme +
                                                    theme(plot.title = element_text(hjust = 0.7, vjust = -1), 
                                                          legend.position = "right", legend.direction = "vertical",
                                                          legend.key.height= unit(4, 'cm'), legend.key.width = unit(2,"cm"),
                                                          legend.text = element_text(size = 38, face = "bold"),
                                                          legend.title = element_text(size = 40, face = "bold"),
                                                          strip.background = element_rect(colour = "grey"), 
                                                          strip.text.y = element_blank(), 
                                                          strip.text.x = element_text(color = "black", size = 44, face = "bold"),
                                                          axis.text.x = element_markdown(size = 44, angle = 45, 
                                                                                         color = "black", face = "bold"),
                                                          axis.text.y = element_blank(),
                                                          axis.ticks.y = element_blank(),
                                                          plot.margin=unit(c(0, 0, 0, -2), "cm"))

#Combine the heatmaps with the coverage data
Long_read_MAGs_sylph_combined_plot <- Starting_material_relative_abundance_heatmap +
                                      Long_read_MAGs_sylph_nitrogen_metabolism_heatmap +
                                      plot_layout(ncol = 2, nrow = 1, widths = c(0.1, 1), guides = "collect")

Long_read_MAGs_sylph_combined_plot
ggsave("Fig1_relative_abundance_and_nitrogen_metabolic_for_starting_consortium_PacBio_sequencing_heatmap.pdf", height=70, width=50, device="pdf", limitsize = FALSE)
```
# Figure S1: Visualize the relative abundance and metabolic data for the PacBio read set generated with different DNA extraction kits used on the starting consortium material

1. Load the long required packages to manipulate the data. Load the long read MAG data that includes relative abundance and taxonomic information for the different DNA kits.
```{r}
#Load the required libraries for data manipulation
library(tidyverse)
library(readxl)
library(data.table)
library(glue)
library(ggtext)
library(ggh4x)
library(patchwork)

#Load the data frame
Long_read_MAG_abundance_DNA_joined_to_taxonomy <- fread("FigS1_relative_abundance_PacBio_DNA_kit_comparison_data.tsv", sep = "\t", header = "auto")

#Write the table to save the source data
write.table(Long_read_MAG_abundance_DNA_joined_to_taxonomy, file = "FigS1_relative_abundance_PacBio_DNA_kit_comparison_data.tsv", sep = "\t", row.names = FALSE)
```

2. Generate a heatmap faceting using the sample name to parse information for DNA kits. The variables in each sample name can be parsed further if need be.
```{r}
heatmap_theme<-  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
                 panel.background = element_blank(), axis.line = element_line(colour = "black"),
                 axis.text.x = element_text(color = "black", size = 14),
                 axis.text.y = element_text(color = "black", size = 14),
                 axis.title.x = element_text(color = "black", size = 16, vjust = -3),
                 axis.title.y = element_text(color = "black", size = 38, vjust = 6),
                 legend.title=element_text(size=32, vjust = 2), 
                 legend.text=element_text(size=30),
                 legend.key = element_rect(fill = "white", color = NA),
                 plot.title = element_text(size = 40, face = "bold", hjust = 0.5),
                 panel.border = element_rect(colour = "black", fill=NA, size=1),
                 plot.margin = unit(c(2, 2, 2, 2), "cm"))

#Create custom order for facets
Long_read_MAG_abundance_DNA_joined_to_taxonomy$DNA_kit = factor(Long_read_MAG_abundance_DNA_joined_to_taxonomy$DNA_kit, 
                                                                        levels =c("MetagenomBio SOX", "Qiagen PowerSoil", "Combined data"))

Long_read_MAG_starting_material_heatmap <- Long_read_MAG_abundance_DNA_joined_to_taxonomy %>%
                                           mutate_at(c('Relative_abundance'), ~na_if(., 0)) %>% #convert 0 to NA to facilitate visualization
                                           filter(!is.na(Relative_abundance)) %>% #need to re add this filter for it to work
                                           ggplot(aes(x= Volume_used,
                                           y = reorder(Family_genus_bin, Relative_abundance), fill= Relative_abundance)) +
                                           geom_tile(aes()) + 
                                           facet_nested(~DNA_kit + Storage, 
                                               scales = "free", space = "free", labeller = labeller(group = label_wrap_gen(10))) +
                                           theme(axis.text.x=element_text(angle=0, hjust=1),
                                           axis.title.y = element_text(color = "black", size = 24, vjust = 0),
                                           legend.key.size = unit(2, 'cm'))+
                                           scale_fill_viridis_c(name="Relative abundance (%)", 
                                                                na.value = "white", limits = c(0, 70)) + #set to max for all families
                                           labs(title = "PacBio MAG abundance from nitrifying cultures subject to different DNA extraction methods",
                                           x = "Volume used (mL)" , y = "MAG ID") +
                                           heatmap_theme +
                                           theme(axis.text.x = element_text(color = "black", size = 32, face = "bold", hjust = 0.5, angle = 0),
                                                 axis.text.y = element_markdown(size = 34, face = "bold"),
                                                 axis.title.x = element_text(color = "black", size = 34, face = "bold"),
                                                 strip.text.x = element_text(color = "black", size = 36, face = "bold"))

Long_read_MAG_starting_material_heatmap
ggsave("FigS1_relative_abundance_for_starting_consortium_PacBio_sequencing_DNA_kit_comparison_heatmap.pdf", height=40, width=50, device="pdf", limitsize = FALSE)
```

# Figure S2: Optical density and pH data for enrichment cultures grown along a redox gradient representative of WWTPs

1. Load the require data analyses packages and the raw data file containing cell growth endpoints and nitrogen analyses.
```{r}
library(readxl)
library(tidyverse)
library(patchwork)

Culture_growth_data <- fread("FigS2_culture_growth_OD_pH_and_nitrogen_analyses_data.tsv", sep = "\t", header = "auto") %>%
                       mutate(Replicate = as.factor(Replicate)) #make sure to convert replicate to a factor for shape coding down the line

#Write the table to save the source data
write.table(Culture_growth_data, file = "FigS2_culture_growth_OD_pH_and_nitrogen_analyses_data.tsv", sep = "\t", row.names = FALSE)
```

2. Generate individual figures for the OD and pH data and then combine them together.
```{r}
theme<- theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        panel.background = element_blank(), axis.line = element_line(colour = "black"),
        axis.text.x = element_text(color = "grey20", 
        size = 24, angle = 0, hjust = .5, vjust = .5, face = "plain"),
        axis.text.y = element_text(color = "grey20", 
        size = 24, angle = 0, hjust = 1, vjust = 0, face = "plain"),  
        axis.title.x = element_text(color = "grey20", 
        size = 26, angle = 0, hjust = .5, vjust = 0, face = "plain"),
        axis.title.y = element_text(color = "grey20", 
        size = 26, angle = 90, face = "plain"),
        strip.text.x = element_text(size = 26, colour = "grey20"),
        strip.text.y = element_text(size = 26, colour = "grey20"),
        plot.title = element_text(size = 30, face = "bold", hjust = .5),
        legend.title=element_text(size=28), 
        legend.text=element_text(size=28),
        panel.border = element_rect(colour = "black", fill=NA, size=1))

#create an ordered version of the metabolism that follows them down the redox ladder
Culture_growth_data$Metabolism_ordered = factor(Culture_growth_data$Metabolism, 
levels =c('Aerobic', 'Nitrate reduction', 'Iron reduction', 'Sulphate reduction'))


OD_scatter_plot<-Culture_growth_data %>%
                ggplot(aes(x=Days_elapsed, y = `OD 600`, color = `Abiotic vs biotic`, shape = Replicate)) +
                geom_point(size = 4) +
                geom_line(aes(group = `Bottle number`, color = `Abiotic vs biotic`), show.legend = FALSE)+
                facet_grid(~ Metabolism_ordered) +
                ylab("O.D. 600 nm") +
                xlab("Days elapsed") + 
                scale_color_brewer(name = "Abiotic vs biotic", palette = "Dark2") +
                theme +
                theme(legend.key = element_blank())

OD_scatter_plot

pH_scatter_plot<-Culture_growth_data %>%
                ggplot(aes(x=Days_elapsed, y = `pH`, color = `Abiotic vs biotic`, shape = Replicate), show.legend = FALSE) +
                geom_point(size = 4, show.legend = FALSE) +
                geom_line(aes(group = `Bottle number`, color = `Abiotic vs biotic`), show.legend = FALSE)+
                facet_grid(~ Metabolism_ordered) +
                ylab("pH") +
                xlab("Days elapsed") + 
                scale_color_brewer(name = "Abiotic vs biotic", palette = "Dark2") +
                theme +
                theme(legend.key = element_blank())

pH_scatter_plot

Redox_scatter_plot<-Culture_growth_data %>%
                ggplot(aes(x=Days_elapsed, y = `Redox`, color = `Abiotic vs biotic`, shape = Replicate)) +
                geom_point(size = 4) +
                geom_line(aes(group = `Bottle number`, color = `Abiotic vs biotic`), show.legend = FALSE)+
                facet_grid(~ Metabolism_ordered) +
                ylab("Redox potential (mV)") +
                xlab("Days elapsed") + 
                scale_color_brewer(name = "Abiotic vs biotic", palette = "Dark2") +
                theme +
                theme(legend.key = element_blank())

Redox_scatter_plot

#Combine the growth data
Growth_data_scatter_plot <-  OD_scatter_plot + pH_scatter_plot + #Redox_scatter_plot +
                             plot_layout(nrow = 2, heights = c(1,1)) +
                             plot_annotation(title = "Optical density and pH from ammonia oxidizing consortia grown along a redox gradient",
                             theme = theme(plot.title = element_text(size = 30, hjust = 0.5, vjust = 0.5, face = "bold")))
               
Growth_data_scatter_plot
ggsave("FigS2_OD_pH_data_from_enrichment_cultures_scatter_plot.pdf", height=12, width=24, device="pdf")
```


# Figure 2: Nitrogen chemistry analyses
1. Load in the data and required packages from your target directory
```{r}
library(readxl)
library(tidyverse)
library(patchwork)

Nitrogen_cycling_data <- fread("Fig2_culture_growth_OD_pH_and_nitrogen_analyses_data.tsv", sep = "\t", header = "auto") %>% #same data as growth but keyed to figure 2
                         mutate(Replicate = as.factor(Replicate)) #make sure to convert replicate to a factor for shape coding down the line

#Write the table to save the source data
write.table(Nitrogen_cycling_data, file = "Fig2_culture_growth_OD_pH_and_nitrogen_analyses_data.tsv", sep = "\t", row.names = FALSE)
```

2. Edit the data to remove day 18 and 0 values.
```{r}
Nitrogen_cycling_data_no_negatives<- Nitrogen_cycling_data %>% filter(!Days_elapsed == 18)

#Remove negative values
Nitrogen_cycling_data_no_negatives[Nitrogen_cycling_data_no_negatives <0] <- 0
```

3. Generate dedicated plots for the 10-fold diluted NH3 and undiluted NO2 measurements conducted with colorimetric methods. Generate an additional plot for the 10-fold diluted NO3- analyses that were run with standard techniques by a certified lab.
```{r}
theme<- theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
        panel.background = element_blank(), axis.line = element_line(colour = "black"),
        axis.text.x = element_text(color = "grey20", 
        size = 24, angle = 0, hjust = .5, vjust = .5, face = "plain"),
        axis.text.y = element_text(color = "grey20", 
        size = 24, angle = 0, hjust = 1, vjust = 0, face = "plain"),  
        axis.title.x = element_text(color = "grey20", 
        size = 26, angle = 0, hjust = .5, vjust = 0, face = "plain"),
        axis.title.y = element_text(color = "grey20", 
        size = 26, angle = 90, face = "plain"),
        strip.text.x = element_text(size = 26, colour = "grey20"),
        strip.text.y = element_text(size = 26, colour = "grey20"),
        plot.title = element_text(size = 30, face = "bold", hjust = .5),
        legend.title=element_text(size=28), 
        legend.text=element_text(size=28),
        panel.border = element_rect(colour = "black", fill=NA, size=1))

#Create a custom order to the facets
Nitrogen_cycling_data_no_negatives$Metabolism_ordered = factor(Nitrogen_cycling_data_no_negatives$Metabolism, 
                                                        levels =c('Aerobic', 'Nitrate reduction', 'Iron reduction', 'Sulphate reduction'))

#Generate 10-fold diluted NH3 plot
NH3_10X_scatter_plot<- Nitrogen_cycling_data_no_negatives %>%
                       ggplot(aes(x=Days_elapsed, y = `10X NH3 concentration mM Carleton`, 
                                  color = `Abiotic vs biotic`, shape = Replicate)) +
                       geom_point(size = 4) +
                       geom_line(aes(group = `Bottle number`, color = `Abiotic vs biotic`))+
                       facet_grid(~ Metabolism_ordered) +
                       ylab("NH3 concentration (mM)") +
                       xlab("Days elapsed") + 
                       scale_color_brewer(name = "Abiotic vs biotic", palette = "Dark2") +
                       theme +
                       theme(legend.position="none")

NH3_10X_scatter_plot

#Generate undiluted NO2 plot
NO2_1X_scatter_plot<- Nitrogen_cycling_data_no_negatives %>%
                      ggplot(aes(x=Days_elapsed, y = `1X NO2 concentration mM Carleton`, 
                                 color = `Abiotic vs biotic`, shape = Replicate)) +
                      geom_point(size = 4) +
                      geom_line(aes(group = `Bottle number`, color = `Abiotic vs biotic`))+
                      facet_grid(~ Metabolism_ordered) +
                      labs(x = xlab("Days elapsed"), y = ylab("NO2- concentration (mM)")) +
                      scale_color_brewer(name = "Abiotic vs biotic", palette = "Dark2") +
                      theme +
                      theme(legend.key=element_blank())

NO2_1X_scatter_plot

#Generaye 10-fold diluted NO3 plot
NO3_10X_scatter_plot <- Nitrogen_cycling_data_no_negatives %>%
                        drop_na(`10X NO3 concentration mM BV`) %>%
                        ggplot(aes(x=Days_elapsed, y = `10X NO3 concentration mM BV`, 
                                   color = `Abiotic vs biotic`, shape = Replicate)) +
                        geom_point(size = 4) +
                        geom_line(aes(group = `Bottle number`, color = `Abiotic vs biotic`))+
                        facet_grid(~ Metabolism_ordered) +
                        labs(x = xlab("Days elapsed"), y = ylab("NO3- concentration (mM)")) +
                        scale_color_brewer(name = "Abiotic vs biotic", palette = "Dark2") +
                        theme +
                        theme(legend.position="none")


NO3_10X_scatter_plot

Final_nitrogen_plot <- NH3_10X_scatter_plot +
                       NO2_1X_scatter_plot +
                       NO3_10X_scatter_plot +
                       plot_layout(nrow = 3, heights = c(1,1)) +
                       plot_annotation(title = "",
                       theme = theme(plot.title = element_text(size = 30, hjust = 0.5, vjust = 0.5, face = "bold")))

Final_nitrogen_plot
ggsave("Fig2_nitrogen_chemical_analyses_scatter_plot.pdf", height=20, width=24, device="pdf")
```
# Figure 3: Beta diversity analyses using non-metric multidimensional scaling for metagenomic DNA sequences obtained from enrichment culture samples

1. Load the required packages and raw data. Manipulate it so it is in the proper format to generate a dissimilarity matrix.
```{r}
library(tidyverse)
library(vegan)
library(data.table)

Full_dereplicated_MAG_abundance_DNA <- fread("Fig3_enrichment_culture_MAG_relative_abundance_data.tsv", sep = "\t", header = "auto")

#Write the table to save the source data
write.table(Full_dereplicated_MAG_abundance_DNA , file = "Fig3_enrichment_culture_MAG_relative_abundance_data.tsv", sep = "\t", row.names = FALSE)

#Need to join this so there is taxonomy data associated because it is already in a matrix shape
#Pivot table so that the sample names are in the row names and we can pull that metadata out afterwards

Full_dereplicated_MAG_abundance_DNA_longer <- Full_dereplicated_MAG_abundance_DNA %>%
                                              pivot_longer(cols = !genome,
                                                           names_to = "Sample",
                                                           values_to = "Relative_abundance")

Full_dereplicated_MAG_abundance_DNA_wider <- Full_dereplicated_MAG_abundance_DNA_longer %>%
                                             pivot_wider(names_from = "genome",
                                                         values_from = "Relative_abundance") %>%
                                             as.data.frame()
                                              

rownames(Full_dereplicated_MAG_abundance_DNA_wider) <- Full_dereplicated_MAG_abundance_DNA_wider$Sample #move sample variable to rowname

Full_dereplicated_MAG_abundance_DNA_wider_no_sample <- Full_dereplicated_MAG_abundance_DNA_wider %>%
                                                       select(-Sample) %>%
                                                       as.data.frame() #need to make sure this is a data frame tibbles don't support row name
```

2. Generate the dissimilarity matrix and the figure, making sure to take note of the raw values output by the stress test for the figure legend.
```{r}
#Generate matrix with bray-curtis dissimilarity index
Full_dereplicated_MAG_abundance_DNA_wider_no_sample <- as.matrix(Full_dereplicated_MAG_abundance_DNA_wider_no_sample)
Full_dereplicated_MAG_abundance_DNA_wider_matrix <- vegdist(Full_dereplicated_MAG_abundance_DNA_wider_no_sample, method = "bray") 

#Generate the NMDS analysis setting the seed to 123 for consistency between runs
set.seed(123)
Full_dereplicated_MAG_NMDS <- metaMDS(Full_dereplicated_MAG_abundance_DNA_wider_matrix)


#Extract scores and convert it to a tibble that can be manipulated in ggplot2
Full_dereplicated_MAG_scores <- scores(Full_dereplicated_MAG_NMDS) %>%
                                as_tibble(rownames = "Sample")

#Parse categorical variables out from the sample names
Full_derpelicates_MAG_scores_categorical_variables <- Full_dereplicated_MAG_scores %>%
                                                      mutate(Time_point = case_when(
                                                      grepl("Day0", Full_dereplicated_MAG_scores$Sample) ~ "Week 0",
                                                      grepl("Week4", Full_dereplicated_MAG_scores$Sample) ~ "Week 4",
                                                      grepl("Week8", Full_dereplicated_MAG_scores$Sample) ~ "Week 8")) %>%
                                                      # Add metabolic treatment variable
                                                      mutate(Metabolic_treatment = case_when(
                                                      grepl("Day0", Full_dereplicated_MAG_scores$Sample) ~ "Inoculum",
                                                      grepl("Aerobic", Full_dereplicated_MAG_scores$Sample) ~ "Aerobic",
                                                      grepl("Ironreduction", Full_dereplicated_MAG_scores$Sample) ~ "Iron reduction",
                                                      grepl("Nitratereduction",Full_dereplicated_MAG_scores$Sample) ~ "Nitrate reduction",
                                                      grepl("Sulphatereduction", Full_dereplicated_MAG_scores$Sample) ~ 
                                                           "Sulphate reduction")) %>%
                                                      #Add replicate number variable
                                                      mutate(Replicate_number = case_when(
                                                      grepl("rep1", Full_dereplicated_MAG_scores$Sample) ~ "1",
                                                      grepl("rep2", Full_dereplicated_MAG_scores$Sample) ~ "2",
                                                      grepl("rep3", Full_dereplicated_MAG_scores$Sample) ~ "3")) %>%
                                                      relocate(c(`Time_point`: `Replicate_number`), .after=Sample)

#Generate the plot

Full_dereplicated_MAG_ordination <-  ggplot(Full_derpelicates_MAG_scores_categorical_variables, aes(x = NMDS1, y = NMDS2)) + 
                                     geom_point(aes(colour = Metabolic_treatment, shape = Time_point), size = 8, alpha = 0.7)+ 
                                     theme(axis.text.y = element_text(colour = "black", size = 12, face = "bold"), 
                                     axis.text.x = element_text(colour = "black", face = "bold", size = 12), 
                                     legend.text = element_text(size = 12, face ="bold", colour ="black"), 
                                     legend.position = "right", axis.title.y = element_text(face = "bold", size = 14), 
                                     axis.title.x = element_text(face = "bold", size = 14, colour = "black"), 
                                     legend.title = element_text(size = 14, colour = "black", face = "bold"), 
                                     panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
                                     legend.key=element_blank()) + 
                                     labs(x = "NMDS1", y = "NMDS2", title = "")

Full_dereplicated_MAG_ordination <- Full_dereplicated_MAG_ordination + 
                                    guides(colour = guide_legend(title ="Metabolic treatment")) + 
                                    guides(shape = guide_legend(title ="Time point"))
 
Full_dereplicated_MAG_ordination
ggsave("Fig3_dereplicated_MAG_enrichment_cultures_time_series_NMDS_ordination.pdf", height=6, width=8, device="pdf", limitsize = FALSE)
```



# Figure S3: Beta diversity analyses using non-metric multidimensional scaling for metagenomic DNA sequences obtained from enrichment culture samples omitting aerobic cultures

1. Load the required packages and raw data. Manipulate it so it is in the proper format to generate a dissimilarity matrix.
```{r}
library(tidyverse)
library(vegan)
library(data.table)

Full_dereplicated_MAG_abundance_DNA <- fread("FigS3_enrichment_culture_MAG_relative_abundance_no_aerobic_cultures_data.tsv", sep = "\t", header = "auto")

#Write the table to save the source data
write.table(Full_dereplicated_MAG_abundance_DNA , file = "FigS3_enrichment_culture_MAG_relative_abundance_no_aerobic_cultures_data.tsv", sep = "\t", row.names = FALSE)

#Remove the aerobic cultures from the raw data
Full_dereplicated_MAG_abundance_DNA_longer_no_aerobic <- Full_dereplicated_MAG_abundance_DNA %>%
                                                         pivot_longer(cols = !genome,
                                                                      names_to = "Sample",
                                                                      values_to = "Relative_abundance") %>%
                                                         filter(!grepl("Aerobic", Sample))

Full_dereplicated_MAG_abundance_DNA_wider_no_aerobic <- Full_dereplicated_MAG_abundance_DNA_longer_no_aerobic %>%
                                                        pivot_wider(names_from = "genome",
                                                                    values_from = "Relative_abundance") %>%
                                                        as.data.frame()

rownames(Full_dereplicated_MAG_abundance_DNA_wider_no_aerobic) <- Full_dereplicated_MAG_abundance_DNA_wider_no_aerobic$Sample

Full_dereplicated_MAG_abundance_DNA_wider_no_aerobic_no_sample <- Full_dereplicated_MAG_abundance_DNA_wider_no_aerobic %>%
                                                       select(-Sample) %>%
                                                       as.data.frame() #need to make sure this is a data frame tibbles don't support row name
```

2. Generate the dissimilarity matrix and the figure, making sure to take note of the raw values output by the stress test for the figure legend.
```{r}
#Generate the dissimilarity matrix
Full_dereplicated_MAG_abundance_DNA_wider_no_aerobic_no_sample <- as.matrix(Full_dereplicated_MAG_abundance_DNA_wider_no_aerobic_no_sample)
Full_dereplicated_MAG_abundance_DNA_wider_no_aerobic_matrix <- vegdist(Full_dereplicated_MAG_abundance_DNA_wider_no_aerobic_no_sample, method = "bray") 

#Generate the NMDS analysis setting the seed to 123 for consistency between runs
set.seed(123)
Full_dereplicated_MAG_no_aerobic_NMDS <- metaMDS(Full_dereplicated_MAG_abundance_DNA_wider_no_aerobic_matrix)


#Extract scores and convert it to a tibble that can be manipulated in ggplot2
Full_dereplicated_MAG_no_aerobic_scores <- scores(Full_dereplicated_MAG_no_aerobic_NMDS) %>%
                                           as_tibble(rownames = "Sample")

#Parse categorical variables out from the sample names
Full_derpelicates_MAG_no_aerobic_scores_categorical_variables <- Full_dereplicated_MAG_no_aerobic_scores %>%
                                                      mutate(Time_point = case_when(
                                                      grepl("Day0", Full_dereplicated_MAG_no_aerobic_scores$Sample) ~ "Week 0",
                                                      grepl("Week4", Full_dereplicated_MAG_no_aerobic_scores$Sample) ~ "Week 4",
                                                      grepl("Week8", Full_dereplicated_MAG_no_aerobic_scores$Sample) ~ "Week 8")) %>%
                                                      # Add metabolic treatment variable
                                                      mutate(Metabolic_treatment = case_when(
                                                      grepl("Day0", Full_dereplicated_MAG_no_aerobic_scores$Sample) ~ "Inoculum",
                                                      #grepl("Aerobic", Full_dereplicated_MAG_no_aerobic_scores$Sample) ~ "Aerobic",
                                                      grepl("Ironreduction", Full_dereplicated_MAG_no_aerobic_scores$Sample) ~ "Iron reduction",
                                                      grepl("Nitratereduction",Full_dereplicated_MAG_no_aerobic_scores$Sample) ~ "Nitrate reduction",
                                                      grepl("Sulphatereduction", Full_dereplicated_MAG_no_aerobic_scores$Sample) ~ 
                                                           "Sulphate reduction")) %>%
                                                      #Add replicate number variable
                                                      mutate(Replicate_number = case_when(
                                                      grepl("rep1", Full_dereplicated_MAG_no_aerobic_scores$Sample) ~ "1",
                                                      grepl("rep2", Full_dereplicated_MAG_no_aerobic_scores$Sample) ~ "2",
                                                      grepl("rep3", Full_dereplicated_MAG_no_aerobic_scores$Sample) ~ "3")) %>%
                                                      relocate(c(`Time_point`: `Replicate_number`), .after=Sample)

#Generate the plot
Full_dereplicated_MAG_no_aerobic_ordination <-  ggplot(Full_derpelicates_MAG_no_aerobic_scores_categorical_variables, aes(x = NMDS1, y = NMDS2)) + 
                                                geom_point(aes(colour = Metabolic_treatment, shape = Time_point), size = 8, alpha = 0.7)+ 
                                                theme(axis.text.y = element_text(colour = "black", size = 12, face = "bold"), 
                                                axis.text.x = element_text(colour = "black", face = "bold", size = 12), 
                                                legend.text = element_text(size = 12, face ="bold", colour ="black"), 
                                                legend.position = "right", axis.title.y = element_text(face = "bold", size = 14), 
                                                axis.title.x = element_text(face = "bold", size = 14, colour = "black"), 
                                                legend.title = element_text(size = 14, colour = "black", face = "bold"), 
                                                panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
                                                legend.key=element_blank()) + 
                                                labs(x = "NMDS1", y = "NMDS2", title = "")

Full_dereplicated_MAG_no_aerobic_ordination <- Full_dereplicated_MAG_no_aerobic_ordination + 
                                    guides(colour = guide_legend(title ="Metabolic treatment")) + 
                                    guides(shape = guide_legend(title ="Time point"))
 
Full_dereplicated_MAG_no_aerobic_ordination

ggsave("FigS3_dereplicated_MAG_enrichment_cultures_time_series_aerobic_cultures_omitted_NMDS_ordination.pdf", height=6, width=8, device="pdf", limitsize = FALSE)
```
# Figure 4: Relative abundance for MAG >= 1% from enrichment cultures grown along a redox gradient

1. Load in the data and required packages from your target directory. 
```{r}
library(tidyverse)
library(readxl)
library(data.table)
library(glue)
library(ggtext)
library(ggh4x)
library(plotly)

Full_dereplicated_time_series_data_formatted_grouping_variables <- fread("Fig4_relative_abundance_and_nitrogen_metabolism_for_Illumina_MAGs_enrichment_cultures_data.tsv", 
                                                                         sep = "\t", header = "auto") %>%
                                                                   mutate(Replicate_number = as.factor(Replicate_number)) #make sure to convert to a factor for heat maps
                                                                    
#Write the table to save the source data
write.table(Full_dereplicated_time_series_data_formatted_grouping_variables, 
            file = "Fig4_relative_abundance_and_nitrogen_metabolism_for_Illumina_MAGs_enrichment_cultures_data.tsv", 
            sep = "\t", row.names = FALSE)
```

2. Make the data into a long format to recreate the DRAM output figures.
```{r}
Full_dereplicated_time_series_percent_complete_gathered <- Full_dereplicated_time_series_data_formatted_grouping_variables %>% 
                                                           gather(key = Pathway_name, value = Percent_complete, 
                                                           "3-Hydroxypropionate bi-cycle":"Complex V: V/A-type ATPase, prokaryotes", na.rm=FALSE) %>%
                                                           select(-c("CAZy: Alpha-galactans": "Sulfur metabolism: thiosulfate => sulfite")) %>% 
                                                           mutate(group = case_when(grepl("\\:.*", Pathway_name) ~ 
                                                           sub("\\:.*", "", Pathway_name),
                                                           !grepl("\\:.*", Pathway_name) ~ "Module")) %>% #maintained up to this step
                                                           mutate(Pathway_name = gsub("Complex I*:","",Pathway_name)) %>%
                                                           mutate(Pathway_name = gsub("Complex IV*.*:","",Pathway_name)) %>%
                                                           mutate(Pathway_name = gsub("Complex V*.*:","",Pathway_name))

#80704 observations for 20 variables

Full_dereplicated_time_series_true_false_gathered <- Full_dereplicated_time_series_data_formatted_grouping_variables %>%
                                                     gather(Pathway_name, Presence_absence,
                                                     "CAZy: Alpha-galactans":"Sulfur metabolism: thiosulfate => sulfite", na.rm = FALSE) %>%
                                                     select(-c("3-Hydroxypropionate bi-cycle":"Complex V: V/A-type ATPase, prokaryotes")) %>%
                                                     mutate(group = case_when(grepl("\\:.*", Pathway_name) ~ 
                                                     sub("\\:.*", "", Pathway_name))) %>%
                                                     mutate(Pathway_name = gsub(".*:","",Pathway_name))

#166452 observations for 20 variables

#Clean up and reorder
Full_dereplicated_time_series_percent_complete_gathered$Pathway_name<-trimws(
                                                                                Full_dereplicated_time_series_percent_complete_gathered$Pathway_name, 
                                                                                which = c("left")) 

Full_dereplicated_time_series_percent_complete_gathered$Pathway_name = factor(
                                          Full_dereplicated_time_series_percent_complete_gathered$Pathway_name, 
                                          levels =c(
                                            
                                          #Module facet
                                            
                                          "Glycolysis (Embden-Meyerhof pathway), glucose => pyruvate", 
                                          "Pentose phosphate pathway (Pentose phosphate cycle)",
                                          "Entner-Doudoroff pathway, glucose-6P => glyceraldehyde-3P + pyruvate",
                                          "Citrate cycle (TCA cycle, Krebs cycle)",
                                          "Glyoxylate cycle",
                                          "Reductive pentose phosphate cycle (Calvin cycle)",
                                          "Reductive citrate cycle (Arnon-Buchanan cycle)",
                                          "Dicarboxylate-hydroxybutyrate cycle",
                                          "Hydroxypropionate-hydroxybutylate cycle",
                                          "3-Hydroxypropionate bi-cycle",
                                          "Reductive acetyl-CoA pathway (Wood-Ljungdahl pathway)",
                                          "Acetyl-CoA pathway, CO2 => acetyl-CoA",
                                          "Methanogenesis, CO2 => methane",
                                          
                                          #Complex I
                                          
                                          "NADH:quinone oxidoreductase, prokaryotes",
                                          "NAD(P)H:quinone oxidoreductase, chloroplasts and cyanobacteria",
                                          "NADH dehydrogenase (ubiquinone) 1 alpha subcomplex",
                                          
                                          #Complex II
                                          
                                          "Succinate dehydrogenase (ubiquinone)",
                                          "Succinate dehydrogenase, prokaryotes",
                                          "Fumarate reductase, prokaryotes",
                                          
                                          #Complex III
                                          
                                          "Cytochrome bc1 complex respiratory unit", #what's the difference between first and second?
                                          "Cytochrome bc1 complex",
                                          "Cytochrome bd ubiquinol oxidase",
                                          
                                          #Complex IV high affinity
                                          
                                          #"Cytochrome bd ubiquinol oxidase", This level is being duplicated, but may not be an issue
                                          "Cytochrome c oxidase, cbb3-type",
                                          
                                          #Complex IV low affinity
                                          
                                          "Cytochrome c oxidase",
                                          "Cytochrome c oxidase, prokaryotes",
                                          "Cytochrome aa3-600 menaquinol oxidase",
                                          "Cytochrome o ubiquinol oxidase",
                                          
                                          #Complex V
                                          "F-type ATPase, prokaryotes and chloroplasts",
                                          "F-type ATPase, eukaryotes",
                                          "V/A-type ATPase, prokaryotes",
                                          "V-type ATPase, eukaryotes"
                                          ))

#Categorical data script
Full_dereplicated_time_series_true_false_gathered$Pathway_name<-trimws(Full_dereplicated_time_series_true_false_gathered$Pathway_name, which = c("left")) 

Full_dereplicated_time_series_true_false_gathered$Pathway_name =
                                          factor(Full_dereplicated_time_series_true_false_gathered$Pathway_name, 
                                          levels =c(
                                            
                                          #CAZy facet
                                            
                                          "Polyphenolics", 
                                          "Crystalline Cellulose",
                                          "Amorphous Cellulose",
                                          "Xyloglucan",
                                          "Xylans",
                                          "Alpha-mannan",
                                          "Beta-mannan",
                                          "Mixed-Linkage glucans",
                                          "Starch",
                                          "Pectin",
                                          "Arabinan",
                                          "Alpha-galactans",
                                          "Beta-galactan (pectic galactan)",
                                          "Chitin",
                                          "Sulf-Polysachharides",
                                          "Mucin",
                                          "Fucose Cleavage",
                                          "Arabinose cleavage",
                                          "Rhamnose cleavage",
                                          
                                          #Nitrogen metabolism
                                          
                                          "ammonia => nitrite",
                                          "Bacterial/Archaeal ammonia oxidation",
                                          "Bacterial (aerobic-specific) ammonia oxidation",
                                          "Bacterial (anaerobic-specific) ammonia oxidation",
                                          "nitrite => nitrate",
                                          "nitrate => nitrite",
                                          "nitrite => nitric oxide",
                                          "nitric oxide => nitrous oxide",
                                          "nitrous oxide => nitrogen",
                                          "nitrogen => ammonia",
                                          "Dissimilatory nitrite reduction to ammonia (DNRA)",
                                          "Nitrogen fixation alternative", #typo corrected higher up
                                          
                                          #Sulfur metabolism
                                          "dissimilatory sulfate reduction (and oxidation) sulfate => sulfide",
                                          "thiosulfate => sulfite",
                                          "Thiosulfate oxidation by SOX complex, thiosulfate => sulfate",
                                          "tetrathionate => thiosulfate",
                                          
                                          #Other reductases
                                          
                                          "selenate/Chlorate reduction",
                                          "TMAO reductase",
                                          "mercury reduction",
                                          "arsenate reduction, pt 1",
                                          "arsenate reduction, pt 2",
                                          
                                          #Photosynthesis
                                          "Photosystem I",
                                          "Photosystem II",
                                          
                                          #Methanogenesis and methanotrophy
                                          
                                          "Key functional gene",
                                          "acetate => methane, pt 1",
                                          "acetate => methane, pt 2",
                                          "acetate => methane, pt 3",
                                          "methanol => methane",
                                          "trimethylamine => dimethylamine",
                                          "dimethylamine => monomethylamine",
                                          "monomethylamine => ammonia",
                                          "putative but not defining CO2 => methane",
                                          "methane => methanol, with oxygen (pmo)",
                                          "methane => methanol, with oxygen (mmo)",
                                          
                                          #SCFA and alcohol conversions
                                          "pyruvate => acetyl CoA v1",
                                          "pyruvate => acetyl CoA v2",
                                          "pyruvate => acetylCoA f+ formate v3",
                                          "acetate, pt 1",
                                          "acetate, pt 2",
                                          "acetate, pt 3",
                                          "lactate L",
                                          "lactate D",
                                          "Butyrate, pt 1",
                                          "Butyrate, pt 2",
                                          "Propionate, pt 1",
                                          "Propionate, pt 2",
                                          "Alcohol production"
                                          ))
```

3. Generate a relative abundance heat map with a >= 1% relative abundance cutoff.
```{r}
#Generate aesthetic theme

heatmap_theme<-  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
                 panel.background = element_blank(), axis.line = element_line(colour = "black"),
                 axis.text.x = element_text(color = "black", size = 14),
                 axis.text.y = element_text(color = "black", size = 14),
                 axis.title.x = element_text(color = "black", size = 16, vjust = -3),
                 axis.title.y = element_text(color = "black", size = 16, vjust = 6),
                 legend.title=element_text(size=40, vjust = 2), 
                 legend.text=element_text(size=38),
                 legend.key = element_rect(fill = "white", color = NA),
                 plot.title = element_text(size = 68, face = "bold", hjust = 0.5),
                 panel.border = element_rect(colour = "black", fill=NA, size=1),
                 plot.margin = unit(c(2, 2, 2, 4), "cm"))

#Create custom order for the grouping variables
Full_dereplicated_time_series_percent_complete_gathered$Time_point = factor(Full_dereplicated_time_series_percent_complete_gathered$Time_point, levels =c("Week 0", "Week 4", "Week 8"))
                                                                        
Full_dereplicated_time_series_percent_complete_gathered$Metabolic_treatment = factor(
Full_dereplicated_time_series_percent_complete_gathered$Metabolic_treatment, 
levels =c("Inoculum",
         "Aerobic",
         "Nitrate reduction",
         "Iron reduction",
         "Sulphate reduction"))

Full_dereplicated_time_series_abundance_heatmap_1_percent_cutoff <- Full_dereplicated_time_series_percent_complete_gathered %>%
                                                                    filter(Relative_abundance >= 1) %>% #add this filter for a 1% cutoff
                                               ggplot(aes(x= Replicate_number,
                                               y = reorder(Family_genus_bin, Relative_abundance), fill= Relative_abundance)) +
                                               geom_tile(aes()) + 
                                               facet_nested(~Metabolic_treatment + Time_point, 
                                               scales = "free", space = "free", labeller = labeller(group = label_wrap_gen(10))) +
                                               theme(axis.text.x=element_text(angle=0, hjust=1),
                                               axis.title.y = element_text(color = "black", size = 24, vjust = 0),
                                               legend.key.size = unit(4, 'cm'))+
                                               scale_fill_viridis_c(name="Relative abundance (%)", 
                                                                    na.value = "white", limits = c(0, 70)) + #set to max which is just above 80
                                               labs(title = "MAG relative abundance >= 1% from nitrifying cultures grown along a redox gradient",
                                               x = "Replicate" , y = "MAG ID") +
                                               heatmap_theme +
                                               theme(axis.text.x = element_text(color = "black", size = 60, face = "bold"),
                                                     axis.text.y = element_markdown(size = 60, face = "bold"),
                                                     axis.title.x = element_text(color = "black", size = 64, face = "bold"),
                                                     axis.title.y = element_text(color = "black", size = 64, vjust = 6, face = "bold"),
                                                     strip.text.x = element_text(color = "black", size = 64, face = "bold"),
                                                     legend.title=element_text(size=64, vjust = 2, face = "bold"), 
                                                     legend.text=element_text(size=62, face = "bold"),
                                                     legend.margin = margin(60,60,60,60))

Full_dereplicated_time_series_abundance_heatmap_1_percent_cutoff
```

4. Generate the gene presence/absence data for nitrogen metabolism as a dedicated plot.
```{r}
heatmap_theme<-  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
                 panel.background = element_blank(), axis.line = element_line(colour = "black"),
                 axis.text.x = element_text(color = "black", size = 14),
                 axis.text.y = element_text(color = "black", size = 14),
                 axis.title.x = element_text(color = "black", size = 16, vjust = -3),
                 axis.title.y = element_text(color = "black", size = 16, vjust = 6),
                 legend.title=element_text(size=16), 
                 legend.text=element_text(size=14),
                 legend.key = element_rect(fill = "white", color = NA),
                 plot.title = element_text(size = 68, face = "bold", hjust = 0.5),
                 panel.border = element_rect(colour = "black", fill=NA, size=1),
                 plot.margin = unit(c(0, 0, 0, 0), "cm"))

#recode the TRUE/FALSe data to presence absence
Full_dereplicated_time_series_true_false_gathered<-Full_dereplicated_time_series_true_false_gathered %>% 
                                           mutate(Presence_absence = as.factor(Presence_absence)) %>%
                                           mutate(Presence_absence = recode(Presence_absence, 'TRUE' = "Present", 'FALSE' = "Absent"))

Full_dereplicated_time_series_nitrogen_metabolism_1_percent_cutoff_heatmap<- Full_dereplicated_time_series_true_false_gathered %>%
                                                    filter(Relative_abundance >= 1) %>% #add cutoff here
                                                    filter(group == "Nitrogen metabolism") %>%
                                                    ggplot(aes(x=Pathway_name, y = reorder(Family_genus_bin, Relative_abundance),
                                                               fill=Presence_absence))+
                                                    geom_tile(aes())+
                                                    facet_grid(~ group, 
                                                             scales = "free", space = "free") +
                                                    theme(axis.text.x=element_text(angle=90, hjust=1)) +
                                                    scale_fill_manual(values = c("#CCFFFF", "#009933"))+ 
                                                    labs(title = "", x = "", y = "", fill = "Function") +
                                                    heatmap_theme +
                                                    theme(plot.title = element_text(hjust = 0.7, vjust = -1), 
                                                          legend.position = "right", legend.direction = "vertical",
                                                          legend.text = element_text(size = 62, face = "bold"),
                                                          legend.title = element_text(size = 64, face = "bold"),
                                                          legend.margin = margin(60, 60, 60, 60),
                                                          legend.key.size = unit(4, 'cm'),
                                                          strip.background = element_rect(colour = "grey"), 
                                                          strip.text.y = element_blank(), 
                                                          strip.text.x = element_text(color = "black", size = 64, face = "bold"),
                                                          axis.text.x = element_markdown(size = 62, angle = 45, color = "black", face = "bold"),
                                                          axis.text.y = element_blank(),
                                                          axis.ticks.y = element_blank(),
                                                          plot.margin=margin(0,0,0,0))

Full_dereplicated_time_series_nitrogen_metabolism_1_percent_cutoff_heatmap
```
5. Combine both plots to create the final figure.
```{r}
#Combine both figures
Full_dereplicated_time_series_1_percent_cutoff_combined_plot <- Full_dereplicated_time_series_abundance_heatmap_1_percent_cutoff +
                                                                Full_dereplicated_time_series_nitrogen_metabolism_1_percent_cutoff_heatmap +
                                                                plot_layout(ncol = 2, nrow = 1, widths = c(1.4, 0.5), guides = "collect")

Full_dereplicated_time_series_1_percent_cutoff_combined_plot
ggsave("Fig4_one_percent_cutoff_relative_abundance_and_nitrogen_metabolism_for_MAG_enrichment_cultures_time_series_heatmap.pdf", 
       height=85, width=110, device="pdf", limitsize = FALSE)
```

# Figure S4: RNA read count data for key nitrogen cycling genes from enrichment cultures

1. Load in the require packages and raw data to generate the figures.
```{r}
library(tidyverse)
library(readxl)
library(data.table)
library(glue)
library(ggtext)
library(ggh4x)
library(plotly)

Nitrogen_redox_read_mapping_data_categories <- fread("FigS4_DNA_and_RNA_read_mapping_for_nitrogen_redox_metabolism_data.tsv", sep = "\t", header = "auto") %>%
                                               mutate(Replicate_number = as.factor(Replicate_number)) #make sure to convert to a factor
                                                                    
#Write the table to save the source data
write.table(Nitrogen_redox_read_mapping_data_categories, file = "FigS4_DNA_and_RNA_read_mapping_for_nitrogen_redox_metabolism_data.tsv", sep = "\t", row.names = FALSE)
```

2. Create a dedicated figure for the RNA data and all of the nitrogen redox cycling pathways  
```{r}
heatmap_theme<-  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
                 panel.background = element_blank(), axis.line = element_line(colour = "black"),
                 axis.text.x = element_text(color = "black", size = 14),
                 axis.text.y = element_text(color = "black", size = 14),
                 axis.title.x = element_text(color = "black", size = 16, vjust = -3),
                 axis.title.y = element_text(color = "black", size = 16, vjust = 6),
                 legend.title=element_text(size=40, vjust = 2), 
                 legend.text=element_text(size=38),
                 legend.key = element_rect(fill = "white", color = NA),
                 plot.title = element_text(size = 68, face = "bold", hjust = 0.5),
                 panel.border = element_rect(colour = "black", fill=NA, size=1),
                 plot.margin = unit(c(4, 4, 4, 4), "cm"))

#Create custom order for the grouping variables
Nitrogen_redox_read_mapping_data_categories$Time_point = factor(Nitrogen_redox_read_mapping_data_categories$Time_point, levels =c("Week 0", "Week 4", "Week 8"))
                                                                        
Nitrogen_redox_read_mapping_data_categories$Metabolic_treatment = factor(Nitrogen_redox_read_mapping_data_categories$Metabolic_treatment, 
                                                                  levels =c("Inoculum",
                                                                           "Aerobic",
                                                                           "Nitrate reduction",
                                                                           "Iron reduction",
                                                                           "Sulphate reduction"))

Nitrogen_redox_read_mapping_data_categories$kegg_ID_custom_order = factor(Nitrogen_redox_read_mapping_data_categories$kegg_ID_custom_order,
                                                                          levels =c(
                                                                          
                                                                           #Ammonia monooxygenase
                                                                           "K10944; methane/ammonia monooxygenase subunit A",
                                                                           "K10945; methane/ammonia monooxygenase subunit B",
                                                                           "K10946; methane/ammonia monooxygenase subunit C",
                                                                           
                                                                           #Hydroxylamine metabolism
                                                                           "K15864; nitrite reductase (NO-forming)/hydroxylamine reductase",
                                                                           "K10535; hydroxylamine dehydrogenase",
                                                                           
                                                                           #Nitrite oxidation (goes both ways)
                                                                           "K00370; nitrate reductase/nitrite oxidoreductase, alpha subunit",
                                                                           "K00371; nitrate reductase/nitrite oxidoreductase, beta subunit",
                                                                           
                                                                           #Nitrate reduction
                                                                           "K00372; assimilatory nitrate reductase catalytic subunit",
                                                                           "K02567; periplasmic nitrate reductase NapA",
                                                                           "K02568; cytochrome c-type protein NapB",
                                                                           
                                                                           #Nitrite reduction"
                                                                           "K00368; nitrite reductase (NO-forming)",
                                                                           "K03385; nitrite reductase (cytochrome c-552)",
                                                                           "K15876; cytochrome c nitrite reductase small subunit",
                                                                           
                                                                           #nitric oxide reduction
                                                                           "K04561; nitric oxide reductase subunit B",
                                                                           
                                                                           #nitrous oxide reduction
                                                                           "K00376; nitrous-oxide reductase"))

Nitrogen_redox_RNA_read_mapping <- Nitrogen_redox_read_mapping_data_categories %>%
                                   filter(DNA_vs_RNA == "RNA") %>%
                                   ggplot(aes(x= Replicate_number, y = kegg_ID_custom_order, fill= TPM)) +
                                   geom_tile(aes()) + 
                                   facet_nested(~Metabolic_treatment + Time_point, 
                                   scales = "fixed", space = "free", labeller = labeller(group = label_wrap_gen(10))) +
                                   scale_y_discrete(limits = rev) +
                                   theme(axis.text.x=element_text(angle=0, hjust=1),
                                   axis.title.y = element_text(color = "black", size = 24, vjust = 0),
                                   legend.key.size = unit(2, 'cm'))+
                                   scale_fill_viridis_c(name="TPM") + #max for DNA is ~ 2000
                                   labs(title = "Read mapping for RNA to nitrogen redox cycling metabolic pathways",
                                   x = "Replicate" , y = "KEGG ID") +
                                   heatmap_theme +
                                   theme(axis.text.x = element_text(color = "black", size = 60, face = "bold"),
                                         axis.text.y = element_markdown(size = 60, face = "bold"),
                                         axis.title.x = element_text(color = "black", size = 64, face = "bold"),
                                         axis.title.y = element_text(color = "black", size = 64, vjust = 0, face = "bold"),
                                         strip.text.x = element_text(color = "black", size = 64, face = "bold"),
                                         legend.title=element_text(size=64, vjust = 2, face = "bold"), 
                                         legend.text=element_text(size=62, face = "bold"))

Nitrogen_redox_RNA_read_mapping
ggsave("FigS4_Nitrogen_redox_metabolism_RNA_read_mapping_heatmap_panelA.pdf", height=35, width=85, device="pdf", limitsize = FALSE) 
```
3. Reproduce the heatmap for RNA counts for nitrogen redox cycling metabolism but omit the Week 0 data
```{r}
Nitrogen_redox_DNA_read_mapping_no_amoC <- Nitrogen_redox_read_mapping_data_categories %>%
                                   filter(DNA_vs_RNA == "RNA") %>%
                                   filter(!Time_point == "Week 0") %>%
                                   ggplot(aes(x= Replicate_number, y = kegg_ID_custom_order, fill= TPM)) +
                                   geom_tile(aes()) + 
                                   facet_nested(~Metabolic_treatment + Time_point, 
                                   scales = "fixed", space = "free", labeller = labeller(group = label_wrap_gen(10))) +
                                   scale_y_discrete(limits = rev) +
                                   theme(axis.text.x=element_text(angle=0, hjust=1),
                                   axis.title.y = element_text(color = "black", size = 24, vjust = 0),
                                   legend.key.size = unit(2, 'cm'))+
                                   scale_fill_viridis_c(name="TPM") + #max for DNA is ~ 2000
                                   labs(title = "Read mapping for RNA to nitrogen redox cycling metabolic pathways",
                                   x = "Replicate" , y = "KEGG ID") +
                                   heatmap_theme +
                                   theme(axis.text.x = element_text(color = "black", size = 60, face = "bold"),
                                         axis.text.y = element_markdown(size = 60, face = "bold"),
                                         axis.title.x = element_text(color = "black", size = 64, face = "bold"),
                                         axis.title.y = element_text(color = "black", size = 64, vjust = 4, face = "bold"),
                                         strip.text.x = element_text(color = "black", size = 64, face = "bold"),
                                         legend.title=element_text(size=64, vjust = 2, face = "bold"), 
                                         legend.text=element_text(size=62, face = "bold"))

Nitrogen_redox_DNA_read_mapping_no_amoC
ggsave("FigS4_Nitrogen_redox_metabolism_RNA_read_mapping_heatmap_panelB.pdf", height=35, width=85, device="pdf", limitsize = FALSE) 
```

# Figure S5: Relative abundance for all MAGs (no abundance cutoff) from enrichment cultures grown along a redox gradient

1. Load in the data and required packages from your target directory. 
```{r}
library(tidyverse)
library(readxl)
library(data.table)
library(glue)
library(ggtext)
library(ggh4x)
library(plotly)

Full_dereplicated_time_series_data_formatted_grouping_variables <- fread("FigS5_relative_abundance_and_nitrogen_metabolism_for_Illumina_MAGs_enrichment_cultures_data.tsv", 
                                                                         sep = "\t", header = "auto") %>%
                                                                   mutate(Replicate_number = as.factor(Replicate_number)) #make sure to convert to a factor
                                                                    
#Write the table to save the source data
write.table(Full_dereplicated_time_series_data_formatted_grouping_variables, 
            file = "FigS5_relative_abundance_and_nitrogen_metabolism_for_Illumina_MAGs_enrichment_cultures_data.tsv", 
            sep = "\t", row.names = FALSE)
```

2. Make the data into a long format to recreate the DRAM output figures.
```{r}
Full_dereplicated_time_series_percent_complete_gathered <- Full_dereplicated_time_series_data_formatted_grouping_variables %>% 
                                                           gather(key = Pathway_name, value = Percent_complete, 
                                                           "3-Hydroxypropionate bi-cycle":"Complex V: V/A-type ATPase, prokaryotes", na.rm=FALSE) %>%
                                                           select(-c("CAZy: Alpha-galactans": "Sulfur metabolism: thiosulfate => sulfite")) %>% 
                                                           mutate(group = case_when(grepl("\\:.*", Pathway_name) ~ 
                                                           sub("\\:.*", "", Pathway_name),
                                                           !grepl("\\:.*", Pathway_name) ~ "Module")) %>% #maintained up to this step
                                                           mutate(Pathway_name = gsub("Complex I*:","",Pathway_name)) %>%
                                                           mutate(Pathway_name = gsub("Complex IV*.*:","",Pathway_name)) %>%
                                                           mutate(Pathway_name = gsub("Complex V*.*:","",Pathway_name))

#80704 observations for 20 variables

Full_dereplicated_time_series_true_false_gathered <- Full_dereplicated_time_series_data_formatted_grouping_variables %>%
                                                     gather(Pathway_name, Presence_absence,
                                                     "CAZy: Alpha-galactans":"Sulfur metabolism: thiosulfate => sulfite", na.rm = FALSE) %>%
                                                     select(-c("3-Hydroxypropionate bi-cycle":"Complex V: V/A-type ATPase, prokaryotes")) %>%
                                                     mutate(group = case_when(grepl("\\:.*", Pathway_name) ~ 
                                                     sub("\\:.*", "", Pathway_name))) %>%
                                                     mutate(Pathway_name = gsub(".*:","",Pathway_name))

#166452 observations for 20 variables

#Clean up and reorder
Full_dereplicated_time_series_percent_complete_gathered$Pathway_name<-trimws(
                                                                                Full_dereplicated_time_series_percent_complete_gathered$Pathway_name, 
                                                                                which = c("left")) 

Full_dereplicated_time_series_percent_complete_gathered$Pathway_name = factor(
                                          Full_dereplicated_time_series_percent_complete_gathered$Pathway_name, 
                                          levels =c(
                                            
                                          #Module facet
                                            
                                          "Glycolysis (Embden-Meyerhof pathway), glucose => pyruvate", 
                                          "Pentose phosphate pathway (Pentose phosphate cycle)",
                                          "Entner-Doudoroff pathway, glucose-6P => glyceraldehyde-3P + pyruvate",
                                          "Citrate cycle (TCA cycle, Krebs cycle)",
                                          "Glyoxylate cycle",
                                          "Reductive pentose phosphate cycle (Calvin cycle)",
                                          "Reductive citrate cycle (Arnon-Buchanan cycle)",
                                          "Dicarboxylate-hydroxybutyrate cycle",
                                          "Hydroxypropionate-hydroxybutylate cycle",
                                          "3-Hydroxypropionate bi-cycle",
                                          "Reductive acetyl-CoA pathway (Wood-Ljungdahl pathway)",
                                          "Acetyl-CoA pathway, CO2 => acetyl-CoA",
                                          "Methanogenesis, CO2 => methane",
                                          
                                          #Complex I
                                          
                                          "NADH:quinone oxidoreductase, prokaryotes",
                                          "NAD(P)H:quinone oxidoreductase, chloroplasts and cyanobacteria",
                                          "NADH dehydrogenase (ubiquinone) 1 alpha subcomplex",
                                          
                                          #Complex II
                                          
                                          "Succinate dehydrogenase (ubiquinone)",
                                          "Succinate dehydrogenase, prokaryotes",
                                          "Fumarate reductase, prokaryotes",
                                          
                                          #Complex III
                                          
                                          "Cytochrome bc1 complex respiratory unit", #what's the difference between first and second?
                                          "Cytochrome bc1 complex",
                                          "Cytochrome bd ubiquinol oxidase",
                                          
                                          #Complex IV high affinity
                                          
                                          #"Cytochrome bd ubiquinol oxidase", This level is being duplicated, but may not be an issue
                                          "Cytochrome c oxidase, cbb3-type",
                                          
                                          #Complex IV low affinity
                                          
                                          "Cytochrome c oxidase",
                                          "Cytochrome c oxidase, prokaryotes",
                                          "Cytochrome aa3-600 menaquinol oxidase",
                                          "Cytochrome o ubiquinol oxidase",
                                          
                                          #Complex V
                                          "F-type ATPase, prokaryotes and chloroplasts",
                                          "F-type ATPase, eukaryotes",
                                          "V/A-type ATPase, prokaryotes",
                                          "V-type ATPase, eukaryotes"
                                          ))

#Categorical data script
Full_dereplicated_time_series_true_false_gathered$Pathway_name<-trimws(Full_dereplicated_time_series_true_false_gathered$Pathway_name, which = c("left")) 

Full_dereplicated_time_series_true_false_gathered$Pathway_name =
                                          factor(Full_dereplicated_time_series_true_false_gathered$Pathway_name, 
                                          levels =c(
                                            
                                          #CAZy facet
                                            
                                          "Polyphenolics", 
                                          "Crystalline Cellulose",
                                          "Amorphous Cellulose",
                                          "Xyloglucan",
                                          "Xylans",
                                          "Alpha-mannan",
                                          "Beta-mannan",
                                          "Mixed-Linkage glucans",
                                          "Starch",
                                          "Pectin",
                                          "Arabinan",
                                          "Alpha-galactans",
                                          "Beta-galactan (pectic galactan)",
                                          "Chitin",
                                          "Sulf-Polysachharides",
                                          "Mucin",
                                          "Fucose Cleavage",
                                          "Arabinose cleavage",
                                          "Rhamnose cleavage",
                                          
                                          #Nitrogen metabolism
                                          
                                          "ammonia => nitrite",
                                          "Bacterial/Archaeal ammonia oxidation",
                                          "Bacterial (aerobic-specific) ammonia oxidation",
                                          "Bacterial (anaerobic-specific) ammonia oxidation",
                                          "nitrite => nitrate",
                                          "nitrate => nitrite",
                                          "nitrite => nitric oxide",
                                          "nitric oxide => nitrous oxide",
                                          "nitrous oxide => nitrogen",
                                          "nitrogen => ammonia",
                                          "Dissimilatory nitrite reduction to ammonia (DNRA)",
                                          "Nitrogen fixation alternative", #typo corrected higher up
                                          
                                          #Sulfur metabolism
                                          "dissimilatory sulfate reduction (and oxidation) sulfate => sulfide",
                                          "thiosulfate => sulfite",
                                          "Thiosulfate oxidation by SOX complex, thiosulfate => sulfate",
                                          "tetrathionate => thiosulfate",
                                          
                                          #Other reductases
                                          
                                          "selenate/Chlorate reduction",
                                          "TMAO reductase",
                                          "mercury reduction",
                                          "arsenate reduction, pt 1",
                                          "arsenate reduction, pt 2",
                                          
                                          #Photosynthesis
                                          "Photosystem I",
                                          "Photosystem II",
                                          
                                          #Methanogenesis and methanotrophy
                                          
                                          "Key functional gene",
                                          "acetate => methane, pt 1",
                                          "acetate => methane, pt 2",
                                          "acetate => methane, pt 3",
                                          "methanol => methane",
                                          "trimethylamine => dimethylamine",
                                          "dimethylamine => monomethylamine",
                                          "monomethylamine => ammonia",
                                          "putative but not defining CO2 => methane",
                                          "methane => methanol, with oxygen (pmo)",
                                          "methane => methanol, with oxygen (mmo)",
                                          
                                          #SCFA and alcohol conversions
                                          "pyruvate => acetyl CoA v1",
                                          "pyruvate => acetyl CoA v2",
                                          "pyruvate => acetylCoA f+ formate v3",
                                          "acetate, pt 1",
                                          "acetate, pt 2",
                                          "acetate, pt 3",
                                          "lactate L",
                                          "lactate D",
                                          "Butyrate, pt 1",
                                          "Butyrate, pt 2",
                                          "Propionate, pt 1",
                                          "Propionate, pt 2",
                                          "Alcohol production"
                                          ))
```

3. Generate a relative abundance heat map without any relative abundance cutoff.
```{r}
#Generate aesthetic theme

heatmap_theme<-  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
                 panel.background = element_blank(), axis.line = element_line(colour = "black"),
                 axis.text.x = element_text(color = "black", size = 14),
                 axis.text.y = element_text(color = "black", size = 14),
                 axis.title.x = element_text(color = "black", size = 16, vjust = -3),
                 axis.title.y = element_text(color = "black", size = 16, vjust = 6),
                 legend.title=element_text(size=40, vjust = 2), 
                 legend.text=element_text(size=38),
                 legend.key = element_rect(fill = "white", color = NA),
                 plot.title = element_text(size = 68, face = "bold", hjust = 0.5),
                 panel.border = element_rect(colour = "black", fill=NA, size=1),
                 plot.margin = unit(c(2, 2, 2, 4), "cm"))

#Create custom order for the grouping variables
Full_dereplicated_time_series_percent_complete_gathered$Time_point = factor(Full_dereplicated_time_series_percent_complete_gathered$Time_point, levels =c("Week 0", "Week 4", "Week 8"))
                                                                        
Full_dereplicated_time_series_percent_complete_gathered$Metabolic_treatment = factor(
Full_dereplicated_time_series_percent_complete_gathered$Metabolic_treatment, 
levels =c("Inoculum",
         "Aerobic",
         "Nitrate reduction",
         "Iron reduction",
         "Sulphate reduction"))

Full_dereplicated_time_series_abundance_heatmap <- Full_dereplicated_time_series_percent_complete_gathered %>%
                                                   mutate_at(c('Relative_abundance'), ~na_if(., 0)) %>% #convert 0 to NA to facilitate visualization
                                                   filter(!is.na(Relative_abundance)) %>% #need to re add this filter for it to work
                                                   ggplot(aes(x= Replicate_number,
                                                   y = reorder(Family_genus_bin, Relative_abundance), fill= Relative_abundance)) +
                                                   geom_tile(aes()) + 
                                                   facet_nested(~Metabolic_treatment + Time_point, 
                                                   scales = "free", space = "free", labeller = labeller(group = label_wrap_gen(10))) +
                                                   theme(axis.text.x=element_text(angle=0, hjust=1),
                                                   axis.title.y = element_text(color = "black", size = 24, vjust = 0),
                                                   legend.key.size = unit(4, 'cm'))+
                                                   scale_fill_viridis_c(name="Relative abundance (%)", 
                                                                    na.value = "white", limits = c(0, 70)) +
                                                   labs(title = "MAG relative abundance from nitrifying cultures grown along a redox gradient",
                                                   x = "Replicate" , y = "MAG ID") +
                                                   heatmap_theme +
                                                   theme(axis.text.x = element_text(color = "black", size = 60, face = "bold"),
                                                         axis.text.y = element_markdown(size = 60, face = "bold"),
                                                         axis.title.x = element_text(color = "black", size = 64, face = "bold"),
                                                         axis.title.y = element_text(color = "black", size = 64, vjust = 6, face = "bold"),
                                                         strip.text.x = element_text(color = "black", size = 64, face = "bold"),
                                                         legend.title=element_text(size=64, vjust = 2, face = "bold"), 
                                                         legend.text=element_text(size=62, face = "bold"),
                                                         legend.margin = margin(60,60,60,60))

Full_dereplicated_time_series_abundance_heatmap
```

4. Generate the gene presence/absence data for nitrogen metabolism as a dedicated plot.
```{r}
heatmap_theme<-  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
                 panel.background = element_blank(), axis.line = element_line(colour = "black"),
                 axis.text.x = element_text(color = "black", size = 14),
                 axis.text.y = element_text(color = "black", size = 14),
                 axis.title.x = element_text(color = "black", size = 16, vjust = -3),
                 axis.title.y = element_text(color = "black", size = 16, vjust = 6),
                 legend.title=element_text(size=16), 
                 legend.text=element_text(size=14),
                 legend.key = element_rect(fill = "white", color = NA),
                 plot.title = element_text(size = 68, face = "bold", hjust = 0.5),
                 panel.border = element_rect(colour = "black", fill=NA, size=1),
                 plot.margin = unit(c(0, 0, 0, 0), "cm"))

#recode the TRUE/FALSe data to presence absence
Full_dereplicated_time_series_true_false_gathered<-Full_dereplicated_time_series_true_false_gathered %>% 
                                           mutate(Presence_absence = as.factor(Presence_absence)) %>%
                                           mutate(Presence_absence = recode(Presence_absence, 'TRUE' = "Present", 'FALSE' = "Absent"))

Full_dereplicated_time_series_nitrogen_metabolism_heatmap<- Full_dereplicated_time_series_true_false_gathered %>%
                                                            filter(group == "Nitrogen metabolism") %>%
                                                            ggplot(aes(x=Pathway_name, y = reorder(Family_genus_bin, Relative_abundance),
                                                                       fill=Presence_absence))+
                                                            geom_tile(aes())+
                                                            facet_grid(~ group, 
                                                                     scales = "free", space = "free") +
                                                            theme(axis.text.x=element_text(angle=90, hjust=1)) +
                                                            scale_fill_manual(values = c("#CCFFFF", "#009933"))+ 
                                                            labs(title = "", x = "", y = "", fill = "Function") +
                                                            heatmap_theme +
                                                            theme(plot.title = element_text(hjust = 0.7, vjust = -1), 
                                                                  legend.position = "right", legend.direction = "vertical",
                                                                  legend.text = element_text(size = 62, face = "bold"),
                                                                  legend.title = element_text(size = 64, face = "bold"),
                                                                  legend.margin = margin(60, 60, 60, 60),
                                                                  legend.key.size = unit(4, 'cm'),
                                                                  strip.background = element_rect(colour = "grey"), 
                                                                  strip.text.y = element_blank(), 
                                                                  strip.text.x = element_text(color = "black", size = 64, face = "bold"),
                                                                  axis.text.x = element_markdown(size = 62, angle = 45, color = "black", face = "bold"),
                                                                  axis.text.y = element_blank(),
                                                                  axis.ticks.y = element_blank(),
                                                                  plot.margin=margin(0,0,0,0))

Full_dereplicated_time_series_nitrogen_metabolism_heatmap
```

5. Combine both plots to create the final figure.
```{r}
#Combine both figures
Full_dereplicated_time_series_combined_plot <- Full_dereplicated_time_series_abundance_heatmap +
                                                                Full_dereplicated_time_series_nitrogen_metabolism_heatmap +
                                                                plot_layout(ncol = 2, nrow = 1, widths = c(1.4, 0.5), guides = "collect")

Full_dereplicated_time_series_combined_plot
ggsave("FigS5_relative_abundance_and_nitrogen_metabolism_for_MAG_enrichment_cultures_time_series_heatmap.pdf", 
       height=100, width=110, device="pdf", limitsize = FALSE)
```

# Figure S6: Percent of RNA reads mapped to genome bins obtained from the incubation time series

1. Load in the data and required packages from your target directory
```{r}
library(tidyverse)
library(readxl)
library(data.table)
library(glue)
library(ggtext)
library(ggh4x)
library(plotly)

RNA_mapping_to_bins_data_categories <- fread("FigS6_RNA_mapping_to_MAGs_data.tsv", sep = "\t", header = "auto")

#Use this to export a tab-delimited file if need be, this codebook uses one that was already formatted and organized for visualization
write.table(RNA_mapping_to_bins_data_categories, file = "FigS6_RNA_mapping_to_MAGs_data.tsv", sep = "\t", row.names = FALSE)  
```

2. Create stacked barplots with a custom order to the facets that mimics the formatting of the heatmaps used for DNA and RNA read counts. Use the color ramp function based on how many bins you have but remember to count one less when deciding on the number of columns across which to split the palette because one will need to be manually defined to be grey for the unmapped reads.
```{r}
#Create a visual theme
barplot_theme<-  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
                 panel.background = element_blank(), axis.line = element_line(colour = "black"),
                 axis.text.x = element_text(color = "black", size = 28),
                 axis.text.y = element_text(color = "black", size = 28),
                 axis.title.x = element_text(color = "black", size = 32, vjust = -3),
                 axis.title.y = element_text(color = "black", size = 32, vjust = 6),
                 strip.text.x = element_text(color = "black", size = 30, face = "bold"),
                 legend.title=element_text(size=40, vjust = 2), 
                 legend.text=element_text(size=34),
                 legend.key = element_rect(fill = "white", color = NA),
                 plot.title = element_text(size = 68, face = "bold", hjust = 0.5),
                 panel.border = element_rect(colour = "black", fill=NA, size=1),
                 plot.margin = unit(c(2, 2, 2, 2), "cm"))


#Create custom order for the grouping variables
RNA_mapping_to_bins_data_categories$Time_point = factor(RNA_mapping_to_bins_data_categories$Time_point, levels =c("Week 0", "Week 4", "Week 8"))
                                                                        
RNA_mapping_to_bins_data_categories$Metabolic_treatment = factor(RNA_mapping_to_bins_data_categories$Metabolic_treatment, 
                                                                  levels =c("Inoculum",
                                                                           "Aerobic",
                                                                           "Nitrate reduction",
                                                                           "Iron reduction",
                                                                           "Sulphate reduction"))

#Prepare a ramped up colour scale to make sure there are enough levels to cover all of the bins
library(RColorBrewer)
nb.cols <- 15
mycolors <- c(colorRampPalette(brewer.pal(15, "Paired"))(nb.cols), 'grey') #do 15 columns in the color ramp and then make the last one grey for unmapped

RNA_mapping_to_bins_barplot <-   RNA_mapping_to_bins_data_categories %>%
                                 ggplot(aes(x= Replicate_number, y = percent, fill = `Genome bin`)) +
                                 geom_bar(position="stack", stat="identity") +
                                 facet_nested(~Metabolic_treatment + Time_point)+
                                 scale_fill_manual(values = mycolors) +
                                 theme(axis.text.x=element_text(angle=0, hjust=1),
                                 axis.title.y = element_text(color = "black", size = 24, vjust = 0),
                                 legend.key.size = unit(2, 'cm'))+
                                 labs(x = "Replicate" , y = "Percent RNA reads mapped (%)") +
                                 barplot_theme

RNA_mapping_to_bins_barplot
ggsave("FigS6_RNA_mapping_to_MAGs_stacked_barplot.pdf", height=20, width=40, device="pdf", limitsize = FALSE)
```
3. Make an interactive version, this requires shrinking down font size considerably for it to fit on a typical monitor.
```{r}
plotly_theme<-  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
                 panel.background = element_blank(), axis.line = element_line(colour = "black"),
                 axis.text.x = element_text(color = "black", size = 14),
                 axis.text.y = element_text(color = "black", size = 14),
                 axis.title.x = element_text(color = "black", size = 16, vjust = -3),
                 axis.title.y = element_text(color = "black", size = 16, vjust = 6),
                 strip.text.x = element_text(color = "black", size = 14, face = "bold"),
                 legend.title=element_text(size=16, vjust = 2), 
                 legend.text=element_text(size=14),
                 legend.key = element_rect(fill = "white", color = NA),
                 panel.border = element_rect(colour = "black", fill=NA, size=1),
                 plot.margin = unit(c(2, 2, 2, 2), "cm"))

RNA_mapping_to_bins_data_categories_labels_trimmed <- RNA_mapping_to_bins_data_categories %>% 
                                                      mutate(Metabolic_treatment = str_replace(string = Metabolic_treatment, pattern = " reduction", replacement = ""))


RNA_mapping_to_bins_data_categories_labels_trimmed$Metabolic_treatment = factor(RNA_mapping_to_bins_data_categories_labels_trimmed$Metabolic_treatment, 
                                                                          levels =c("Inoculum",
                                                                                   "Aerobic",
                                                                                   "Nitrate",
                                                                                   "Iron",
                                                                                   "Sulphate"))

RNA_mapping_to_bins_barplot_plotly <-  RNA_mapping_to_bins_data_categories_labels_trimmed %>%
                                       ggplot(aes(x= Replicate_number, y = percent, fill = `Genome bin`)) +
                                       geom_bar(position="stack", stat="identity") +
                                       facet_nested(~Metabolic_treatment + Time_point,
                                                    scales = "fixed", space = "free", 
                                                    labeller = labeller(group = label_wrap_gen(width = 8)))+
                                       scale_fill_manual(values = mycolors) +
                                       theme(axis.text.x=element_text(angle=0, hjust=1),
                                       axis.title.y = element_text(color = "black", size = 24, vjust = 0),
                                       legend.key.size = unit(2, 'cm'))+
                                       labs(x = "Replicate" , y = "Percent RNA reads mapped (%)") +
                                       plotly_theme

library(plotly)
RNA_mapping_to_bins_barplot_interactive <- ggplotly(RNA_mapping_to_bins_barplot_plotly) %>% 
                                           layout(autosize = F, width = 1700, height = 900, 
                                           showlegend = TRUE, legend=list(font = list(size = 20)),
                                           facet = list(title = list(
                                           xaxis = list(
      title = list(
        text = "Nitrate <br>reduction",  # Multi-line facet label for the x-axis
        font = list(size = 14)
      )
    ))))

library(htmlwidgets)
saveWidget(RNA_mapping_to_bins_barplot_interactive, "FigS6_RNA_mapping_to_MAGs_stacked_barplot.html", selfcontained = F, libdir = "lib")
```
# Figure S7: Top 25 values for RNA reads mapped to amino acid metabolic pathways

1. Load in the tab delimited file containing the 3 amino acid pathways within KEGG.
```{r}
library(tidyverse)
library(readxl)
library(data.table)
library(glue)
library(ggtext)
library(ggh4x) #packages for making nested facets

Amino_acid_read_mapping_data <- fread("FigS7_RNA_read_mapping_for_amino_acid_metabolism_top25_hits_data.tsv", sep = "\t", header = "auto")
```

2. Create custom orders for facets and generate heat maps for RNA counts.
```{r}
#Generate theme and custom order for labels

heatmap_theme<-  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
                 panel.background = element_blank(), axis.line = element_line(colour = "black"),
                 axis.text.x = element_text(color = "black", size = 14),
                 axis.text.y = element_text(color = "black", size = 14),
                 axis.title.x = element_text(color = "black", size = 16, vjust = -3),
                 axis.title.y = element_text(color = "black", size = 16, vjust = 6),
                 legend.title=element_text(size=40, vjust = 2), 
                 legend.text=element_text(size=38),
                 legend.key = element_rect(fill = "white", color = NA),
                 plot.title = element_text(size = 68, face = "bold", hjust = 0.5),
                 panel.border = element_rect(colour = "black", fill=NA, size=1),
                 plot.margin = unit(c(2, 2, 2, 2), "cm"))

#Create custom order for the grouping variables
Amino_acid_read_mapping_data$Time_point = factor(Amino_acid_read_mapping_data$Time_point, levels =c("Week 0", "Week 4", "Week 8"))
                                                                        
Amino_acid_read_mapping_data$Metabolic_treatment = factor(Amino_acid_read_mapping_data$Metabolic_treatment, 
                                                          levels =c("Inoculum",
                                                                   "Aerobic",
                                                                   "Nitrate reduction",
                                                                   "Iron reduction",
                                                                   "Sulphate reduction"))

Amino_acid_RNA_read_mapping_top_25 <-  Amino_acid_read_mapping_data %>%
                                       filter(DNA_vs_RNA == "RNA") %>%
                                       ggplot(aes(x= Replicate_number, y = reorder(kegg_ID, desc(TPM)), fill= TPM)) +
                                       geom_tile(aes()) + 
                                       facet_nested(~Metabolic_treatment + Time_point, 
                                       scales = "fixed", space = "free", labeller = labeller(group = label_wrap_gen(10))) +
                                       scale_y_discrete(limits = rev) +
                                       theme(axis.text.x=element_text(angle=0, hjust=1),
                                       axis.title.y = element_text(color = "black", size = 24, vjust = 0),
                                       legend.key.size = unit(2, 'cm'))+
                                       scale_fill_viridis_c(name="TPM") + #max for DNA is ~ 2000
                                       labs(title = "RNA read mapping for amino acid pathways that fell within the top 25 values",
                                       x = "Replicate" , y = "KEGG ID") +
                                       heatmap_theme +
                                       theme(axis.text.x = element_text(color = "black", size = 60, face = "bold"),
                                             axis.text.y = element_markdown(size = 60, face = "bold"),
                                             axis.title.x = element_text(color = "black", size = 64, face = "bold"),
                                             axis.title.y = element_text(color = "black", size = 64, vjust = 0, face = "bold"),
                                             strip.text.x = element_text(color = "black", size = 64, face = "bold"),
                                             legend.title=element_text(size=64, vjust = 2, face = "bold"), 
                                             legend.text=element_text(size=62, face = "bold"))

Amino_acid_RNA_read_mapping_top_25

ggsave("FigS7_amino_acid_RNA_read_mapping_top_25_hits_heatmap.pdf", height=40, width=120, device="pdf", limitsize = FALSE)
```

